src/main/java/edu/agh/wfiis/designpatterns/DesignPatternService.java

```diff
@@ -0,0 +1,55 @@
+package edu.agh.wfiis.designpatterns;
+
+import edu.agh.wfiis.designpatterns.patterns.DesignPattern;
+import edu.agh.wfiis.designpatterns.extractor.ClassPathExtractor;
+import edu.agh.wfiis.designpatterns.extractor.Extractable;
+import edu.agh.wfiis.designpatterns.extractor.FileExtractor;
+import edu.agh.wfiis.designpatterns.enums.PatternName;
+
+import java.io.File;
+
+public class DesignPatternService {
+
+    private Extractable extractor;
+
+    public String provideUsageExample(DesignPattern pattern){
+        return pattern.getExample();
+    }
+
+    public String refactorCode(DesignPattern pattern, String badCode){
+        return pattern.refactor(badCode);
+    }
+
+    public String listPatterns(){
+
+        StringBuilder patternNames = new StringBuilder();
+
+        for(PatternName patternName : PatternName.values())
+        {
+            patternNames.append(patternName.name());
+        }
+
+        return patternNames.toString();
+    }
+
+    public boolean areSameType(DesignPattern pattern1, DesignPattern pattern2){
+        return pattern1.getDesignPatternType().equals(pattern2.getDesignPatternType());
```

Comment:
If you dont treat `DesignPattern` as DTO, this functionality should be in that type.